### PR TITLE
BREAKING CHANGE: add base dependency config and standardize env variable naming

### PR DESCRIPTION
**Pull request by martin-trajanovski**
## Description
This PR introduces a breaking change by adding a base dependency configuration and standardizing the environment variable naming across the application.

## Motivation and Context
This change was necessary to improve the maintainability and consistency of the codebase. By standardizing the environment variable naming, we make the code easier to read and understand, and by adding a base dependency configuration, we ensure that the dependencies across different environments are consistent and manageable.

## Changes
1. Standardized environment variable naming: Previous environment variable names like 'secret', 'tokenLife', and 'baseURL' have been changed to 'JWT_SECRET', 'JWT_TOKEN_LIFE', and 'BASE_URL' respectively. This change has been made across multiple files including '.github/workflows/test-build.yml', 'apps/backend/docker-compose-stfc.e2e.yml', and 'apps/backend/example.development.env'.
2. Added a base dependency configuration: This new configuration has been added in 'apps/backend/src/config/dependencyConfigBase.ts'. It includes the import of various datasources and the setting of the logger.
3. Introduced a new function `configureBaseEnvironment` in 'apps/backend/src/config/base/configureBaseEnvironment.ts' that sets timezone, date time formats and updates OIDC settings. 

These changes improve the consistency, readability, and maintainability of the codebase, thus making it easier for future development and debugging.

## Fixes Jira Issue

[https://jira.esss.lu.se/browse/SWAP-3872](https://jira.esss.lu.se/browse/SWAP-3872)

## Tests included/Docs Updated?

- [ ] I have added tests to cover my changes.
- [ ] All relevant doc has been updated

### CODE REVIEW

**Comment by bolmsten** [comment]
can we make this the configuration by default for the docker-compose and update the readme text for the repo

File: apps/backend/src/config/index.ts [hunk 1]
```diff
@@ -1,6 +1,9 @@
 import 'reflect-metadata';
 
 switch (process.env.DEPENDENCY_CONFIG) {
+  case 'base':
+    require('./dependencyConfigBase');
+    break;
```
Comment (Junjiequan) [inline apps/backend/src/config/index.ts:6]: Why don't we use base config by default if no value is given?

File: apps/backend/src/config/index.ts [hunk 2]
```diff
@@ -16,7 +19,7 @@ switch (process.env.DEPENDENCY_CONFIG) {
   default:
     throw new Error(
       `process.env.DEPENDENCY_CONFIG contains invalid value '${process.env.DEPENDENCY_CONFIG}'.
-       Available values are <e2e|ess|stfc|test>`
+       Available values are <base|e2e|ess|stfc|test>`
```
Comment (Junjiequan) [inline apps/backend/src/config/index.ts:22]: We don't need to throw error by default if we have base config

File: apps/backend/src/config/index.ts (hunk 1 above)
Comment (martin-trajanovski) [reply apps/backend/src/config/index.ts:6]: Maybe this is a good point. I will do the changes and discuss this additionally with STFC if they are OK with this way of having the base config as default.

**Review by Junjiequan — APPROVED** [review]
Looks good to me

**Review by mutambaraf — APPROVED** [review]
Looks good except the comment.

**Pull request merged**